File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        j = len(mat) - 1
        count = 0
        for i in range(len(mat)):
            print(i, j)
            count += mat[i][i] + mat[i][j - i]
        return count if len(mat) % 2 == 0 else count - mat[int(j/2)][int(j/2)] 

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function(mat) {
    if(mat.length===1) return mat[0][0]
    let res=0
    let N=mat.length
    for(let i=0;i<N;i++){
        for(let j=0; j<N;j++){
            if(i===j) res+=mat[i][j]
            else if(i+j===N-1) res+=mat[i][j]
        }
    }
    return res
    
};